storage/objects/download_byte_range.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package objects
+
+// [START storage_download_byte_range]
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"time"
+
+	"cloud.google.com/go/storage"
+)
+
+// downloadByteRange downloads a specific byte range of an object to a file.
+func downloadByteRange(w io.Writer, bucket, object string, startByte int64, length int64, destFileName string) error {
+	// bucket := "bucket-name"
+	// object := "object-name"
+	// startByte := 0
+	// length := 20
+	// destFileName := "file.txt"
+	ctx := context.Background()
+	client, err := storage.NewClient(ctx)
+	if err != nil {
+		return fmt.Errorf("storage.NewClient: %v", err)
+	}
+	defer client.Close()
+
+	ctx, cancel := context.WithTimeout(ctx, time.Second*50)
+	defer cancel()
+
+	f, err := os.Create(destFileName)
+	if err != nil {
+		return fmt.Errorf("os.Create: %v", err)
+	}
```

Comment:
The file should be closed at the end of the sample as well. In this case, you shouldn't use a defer statement because we need to check for errors in writing the file. Instead just do it manually after L58.

I may have forgotten about this in the previous sample, could you double-check that one as well and fix if necessary?